myia/operations/prim_random_uniform.py

```diff
@@ -0,0 +1,58 @@
+"""Definitions for the primitive `random_uniform`."""
+
+from .. import xtype
+from ..lib import (
+    ANYTHING,
+    SHAPE,
+    TYPE,
+    VALUE,
+    AbstractArray,
+    AbstractRandomState,
+    AbstractScalar,
+    AbstractTuple,
+    standard_prim,
+)
+from . import primitives as P
+
+
+def pyimpl_random_uniform(rstate, low, high, size):
+    """Implement `random_uniform`."""
+    value = rstate.uniform(low, high, size)
+    return rstate, value
+
+
+@standard_prim(P.random_uniform)
+async def infer_random_uniform(self, engine,
+                               rstate: AbstractRandomState,
+                               low: AbstractScalar,
+                               high: AbstractScalar,
+                               size: AbstractTuple):
+    """Infer the return type of primitive `random_uniform`."""
+    output_scalar_type = AbstractScalar({VALUE: ANYTHING, TYPE: xtype.f32})
+    output_shape = tuple(
+        self.require_constant(e, argnum=f'"3:size[{edx}]"')
+        for edx, e in enumerate(size.elements))
+    if output_shape:
+        value_type = AbstractArray(output_scalar_type,
+                                   {SHAPE: output_shape, TYPE: xtype.NDArray})
+    else:
+        value_type = output_scalar_type
```

Comment:
It does not seem possible to generate a random array with shape `()`. In general, operations should be as regular as possible, so if the size is a tuple, the output should be an array with that shape. If the user wants a scalar they can use `array_to_scalar`.

Reply:
OK. So I should check that given shape is not an empty tuple ? Because it's valid in pytorch:
```
>>> import torch
>>> torch.zeros(())
tensor(0.)
>>> torch.zeros(()).uniform_()
tensor(0.9694)
```

Reply:
No, it's fine if the shape is the empty tuple, but it should return an AbstractArray in that case (with shape == ()). Basically you can just remove the conditional.

Reply:
Ok!